core/src/main/scala/org/apache/spark/scheduler/StageInfo.scala

```diff
@@ -56,6 +56,8 @@ class StageInfo(
     completionTime = Some(System.currentTimeMillis)
   }
 
+  def attemptNumber(): Int = attemptId
```

Comment:
how about
```
class StageInfo(
    ...
    val attempNumber: Int,
   ...  {
  @deprecated
  def attempId: Int = attemptNumber
}
```

Reply:
If we go this way, I believe it would break source compatibility for Developer API.

Code like `new StageInfo(stageId = xx, attemptId = yy, ...)` couldn't by compiled any more.

Not sure about binary compatibility

Reply:
ah i see, let's keep it.

Reply:
hmmm, can we add a new constructor for `attemptId`?

Reply:
hmmmm, maybe.. Let me try it out after work...

Reply:
It may break the json protocol?

Reply:
So we do not fully deprecate it, it's still `attemptId` in json. Shall we fix the json too? We may need to add some logic at the parser side to recognize `attemptId`

Reply:
So, keep it this way or add a constructor and fix Json parser?

Reply:
I'd like to add a constructor and fix the json parser, otherwise we are not fully deprecating it. But let's do it in a new PR, as it may need quite a lot of changes.

Reply:
Oh, I was wrong. It seems the rest api doesn't use this class. It's using `org.apache.spark.status.api.v1.StageData`. However, it's also called `attemptId` in this class.

Reply:
> But let's do it in a new PR, as it may need quite a lot of changes.

Lets's do that after Spark 2.3 release then? 

>However, it's also called attemptId in this class.

Yeah, (stage)attemptId is over a lot of places...